src/app/shared/services/app-config/app-config.service.ts

```diff
@@ -0,0 +1,43 @@
+import { Injectable } from "@angular/core";
+import { getDefaultAppConfig, IAppConfig, IAppConfigOverride } from "data-models";
+import { BehaviorSubject } from "rxjs";
+import { environment } from "src/environments/environment";
+import { deepMergeObjects } from "../../utils";
+import clone from "clone";
+
+@Injectable({
+  providedIn: "root",
+})
+export class AppConfigService {
+  deploymentOverrides: IAppConfigOverride = (environment.deploymentConfig as any).app_config || {};
+  /** List of constants provided by data-models combined with deployment-specific overrides and skin-specific overrides */
+  appConfig$ = new BehaviorSubject<IAppConfig | {}>({});
+  /** Static value to be read by methods that are not responsive to changes
+   * TODO: migrate all consumers of this value to be responsive to appConfig$ */
+  APP_CONFIG: IAppConfig;
+  deploymentAppConfig: IAppConfig;
+
+  constructor() {
```

Comment:
I see you've now moved the initialisation to the constructor which I think makes sense and should hopefully avoid some of the issues flagged previously